### ISSUE TITLE
Error Hallway Example, Agents don't move when starts training

### ISSUE THREAD

**Issue by moonsh** [Unity-Technologies/ml-agents#1596]
Hi, 

After I succeeded the 3D ball example, I tried the Hallway (not visual hallway) example.

I didn't change anything. I just used the same approach I did like 3D ball example.

Whenever I started training, all agents didn't move suddenly just right after I pushed the play button 1 second later. 

Thank you,

**Comment by cxbxmxcx** [comment]
I believe I am encountering the same issue.

Have updated all my ML-Agents code as of today and still see no improvement.  

What I see is the agent run once and then stops.  I debugged the code and can confirm Move is never called in the agent code.

I have seen this issue for several other example environments.  Running Unity 2018.3.0f2

I can get GridWorld and 3DBall appear to work without issue.

Is there another patch to apply?

Cheers

**Comment by moonsh** [comment]
@vincentpierre You mean the hotfix-v0.6.0a-curricula? and what should I update for my python?  Should I update all packages in the ml-agents environment? Could you explain more about the update?

Thank you,

**Comment by vincentpierre** [comment]
@moonsh The master branch is currently at v0.6.0a, if you cloned the repo, you can update the files by calling `git pull`.
@cxbxmxcx I think this might be a different bug. Can you try in editor training and report any error message you would see in Unity or Python. Also, did you make sure you had your agents set with a Learning Brain ?

**Comment by moonsh** [comment]
@vincentpierre  I think he (@cxbxmxcx ) has the same issue like me. I've updated python but there is no change. Still, agents don't move.

I checked the action value and it showed 0 every step so there was no move
Thank you,
I am using Window and Unity 4.17f,

**Comment by vincentpierre** [comment]
I do not know how to help since I am unable to reproduce this bug. Could you try with a player brain see if the agent can move. Also, It would maybe help to Debug.Log the vector action of the agent in AgentAction.

**Comment by cxbxmxcx** [comment]
Hey Vincent,

Added Debug.Log in the MoveAgent method as you suggested:
int action = Mathf.FloorToInt(act[0]);
**Debug.Log("Action of " + action);**
 switch (action)

Around the second episode of training the agent switches to just picking action 5, which is not an action.

I thought it was happening after an episode, but it looks like it happens during the second episode of training.

Not sure what else to comment about my environment.  I have tried this on a couple Windows 10 machines all running the 2018.3.0f2.

Let me know if you want any other samples run or other logs.  I have a way to debug the Python code in Visual Studio, I will try that next but the TF stuff is hard to track down.

Thanks for the help.

Cheers,

Micheal

**Comment by moonsh** [comment]
@vincentpierre 
I checked agents move well with the player brain and I got same response as Michael said.

I think use_recurrent makes this problem. 

Other environments which are not using the recurrent work fine.

Also, I have a problem with visual learning. Whenever I tried visual learning, it said "Vector observation size mismatch between continuous agent Agent and brain Visual_Learning. Was expecting 1 but received 0".

Thank you,
